internal/nvmeof/nvmeof.go

```diff
@@ -0,0 +1,356 @@
+package nvmeof
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	sync "sync"
+	"time"
+
+	"github.com/ceph/ceph-csi/internal/util/log"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/proto"
+)
+
+// Config holds gateway client configuration.
+type GatewayConfig struct {
+	Address        string
+	Port           string
+	ConnectTimeout int // seconds
```

Comment:
Make attributes as accurate as possible. `Port` would probably be an `int`, and `ConnectTimeout` a `time.Duration`.